### DISCUSSION TITLE
Complete Candle after how many Seconds?

### DISCUSSION THREAD

**Discussion by techietrader** [s4w3d0ff/python-poloniex#133] [Q&A]
Sorry for little off-topic.
I wanted to know the exact seconds after which we get a compelte candle for a period

For instance if I m using 300 seconds (5 min) candle, after how many seconds past 300 seconds should i call there returnChartData to get fully grown candle.

I tried 30 seconds but sometimes I dont get a proper 'close' figure and it gets messed up

This is something I am using to call data every 300 seconds-

```
while True:
    if datetime.datetime.now().minute%5==0:
        time.sleep(30) 
  #Than goes the returnChartData call
```
The above func will call there api after every 5minutes and 30 seconds.
Calling too late can give me data of new forming candle!!
Can someone give me exact figure?

**Comment by s4w3d0ff** [comment]
I just updated the chart example to always update the last candle when called. https://github.com/s4w3d0ff/python-poloniex/commit/1571641590b00159acd211e193e80f271e09d7b3
```
DEBUG:__main__:Getting chart data from db
                         open     close  percentChange
date                                                  
2017-06-29 17:15:00  0.120388  0.119605      -0.006843
2017-06-29 17:20:00  0.119605  0.120272       0.005575
2017-06-29 17:25:00  0.120272  0.120320       0.000401
s4w3d0ff@8core/m/s/E/P/p/e/chart> python3 chart.py
DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): poloniex.com
DEBUG:urllib3.connectionpool:https://poloniex.com:443 "GET /public?end=1498757344.784451&period=300&start=1498757100&currencyPair=BTC_ETH&command=returnChartData HTTP/1.1" 200 None
INFO:__main__:Updating BTC_ETH-300 with 1 new entrys!
1/1 complete 
DEBUG:__main__:Getting chart data from db
                         open     close  percentChange
date                                                  
2017-06-29 17:15:00  0.120388  0.119605      -0.006843
2017-06-29 17:20:00  0.119605  0.120272       0.005575
2017-06-29 17:25:00  0.120272  0.120080      -0.001594
```

And it still updates the forming candle when updating with multiple candles:

```
s4w3d0ff@8core/m/s/E/P/p/e/chart> python3 chart.py
DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): poloniex.com
DEBUG:urllib3.connectionpool:https://poloniex.com:443 "GET /public?period=300&start=1498757100&command=returnChartData&end=1498757803.3182387&currencyPair=BTC_ETH HTTP/1.1" 200 None
INFO:__main__:Updating BTC_ETH-300 with 3 new entrys!
3/3 complete 
DEBUG:__main__:Getting chart data from db
                         open     close  percentChange
date                                                  
2017-06-29 17:25:00  0.120272  0.120000      -0.002259
2017-06-29 17:30:00  0.120000  0.119599      -0.003342
2017-06-29 17:35:00  0.119500  0.119600       0.000008
```

If you don't want the forming candle you could just drop it before you make any calculations or whatever.

I know this doesn't answer your question but it may solve your problem.

I think only poloniex would know what the update time is on the candles they send out, and my guess is that the time would fluctuate. So either constantly updating the candle or dropping it is probably going to be your best option.

**Comment by techietrader** [comment]
Yes. Dropping should work perfectly
Something like.
`df[:-1]`


Thanks mate!
And I tried it for 5 mins data. I called 60 seconds after 5 mins and it drops the forming candle.
